libparsec/crates/tests_macros/src/lib.rs

```diff
@@ -23,24 +28,8 @@ pub fn parsec_test(attr: TokenStream, item: TokenStream) -> TokenStream {
 
         if let Some(testbed) = attributes.testbed {
             // If `testbed` is found, `env: &TestbedEnv` must be set in args
-            match sig.inputs.pop().map(Pair::into_value) {
-                Some(FnArg::Typed(typed)) => {
-                    match typed.pat.as_ref() {
-                        Pat::Ident(pat) if pat.ident == "env" => (),
-                        _ => panic!("The last argument must be: `env: &TestbedEnv`"),
-                    }
-
-                    match typed.ty.as_ref() {
-                        Type::Reference(TypeReference { elem: e, .. }) => match e.as_ref() {
-                            Type::Path(p)
-                                if p.path.segments.last().expect("Incomplete path").ident
-                                    == "TestbedEnv" => {}
-                            _ => panic!("The last argument must be: `env: &TestbedEnv`"),
-                        },
-                        _ => panic!("The last argument must be: `env: &TestbedEnv`"),
-                    }
-                }
-                _ => panic!("Missing argument: `env: &TestbedEnv`"),
+            if let ControlFlow::Break(value) = validate_testbed_argument(&mut sig) {
```

Comment:
Using a good'old `Result` seems more logical here since we stop because the there is errors !

Reply:
It was more logical for me to use ControlFlow to control the flow of the function